exporter/prometheusremotewriteexporter/exporter_test.go

```diff
@@ -1049,3 +1049,67 @@ func TestNoRetryOn4xx(t *testing.T) {
 	assert.True(t, consumererror.IsPermanent(err))
 	assert.Equal(t, 1, attempts)
 }
+
+// Simulate a timeout error that would result in an `http.client` returning an error.
+// This test retries once, and then on the second retry the server is configured to timeout.
+// This should lead to a permanent error.
+func TestNoRetryOnHttpClientError(t *testing.T) {
+	attempts := 0
+	responseHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if attempts < 1 {
+			attempts++
+			http.Error(w, "Bad Request", http.StatusInternalServerError)
+		} else {
+			// Add waiting time for response
+			time.Sleep(20 * time.Millisecond)
+		}
+
+	})
+	mockServer := httptest.NewServer(responseHandler)
+	defer mockServer.Close()
+
+	endpointURL, err := url.Parse(mockServer.URL)
+	require.NoError(t, err)
+
+	settings := &confighttp.HTTPClientSettings{
+		Timeout: 10 * time.Millisecond,
+	}
+	// Create the prwExporter
+	exporterTest := &prwExporter{
```

Comment:
are we actually retrying at all? by instantiating the prw like this, don't we get a configuration initialized with zero values? https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/exporter.go#L38

Reply:
I can see the assertion on line 1090, so we are def retrying. I think we might have found another issue. the component is not respecting the `disable` setting for retrying.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/27592